CMSgov/managed-care-review
# Add Unlock submission functionality

## Summary

This PR covers adding the ability for a CMS user to unlock a submission and no more. CMS users can unlock a submission from the submission summary page, which creates a new revision in the DB. A state user can then edit that new revision using the same old fetch and update DraftSubmission GQL operations. 

Doing this work has helped me develop a design for the next chunk of work. I don’t think that status info like who unlocked or resubmitted a submission belongs in the submission form data models, so I think we need a new graphql endpoint to get that info, and that work will be in a follow on PR. 

NOTE: the text field in the unlock confirmation dialog is not connected to anything yet.

#### Related issues

https://qmacbis.atlassian.net/browse/OY2-12184

#### Screenshots


https://user-images.githubusercontent.com/55759/154122497-32bc5b71-4a40-436e-895c-2baf3c6b16bb.mov


#### Test cases covered
* added API tests for unlock and re-submit
* added FE tests for clicking the unlock button and getting success or errors

## QA guidance
* It’s a known issue that after clicking unlock there is no visual indication that anything worked and if you refresh the page it will error. We need to add some new API to allow CMS users to see unlocked submissions (in the draft state) 
* You should be able to log in as a state user and submit, then as a CMS user and unlock, then as a state user and submit again.
